web-client/src/pages/routes/DashboardRoute.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import Location from 'react-app-location';
+import { Route, RouteProps } from 'react-router-dom';
+import { MenuItem } from 'src/types/menu-item';
+
+import DashboardLayout from '../../components/DashboardLayout/DashboardLayout';
+
+// TODO remove when using real profile state
+const mockProfileData = {
+  fullName: 'John Watkins',
+  photoUrl:
+    'https://lh3.googleusercontent.com/-6DkdLho7c5Y/AAAAAAAAAAI/AAAAAAAAAAA/AMZuuckYv30Z33APqrrucaWdEBbadI7WxQ/photo.jpg?sz=46',
+  followers: 15,
+  stars: 4.5,
+};
+
+// TODO change locations
+const siteLocations = {
+  logout: new Location('/requests/logout'),
+  contact: new Location('/requests/contact'),
+  map: new Location('/requests/map'),
+  notifications: new Location('requests/notifications'),
+};
+
+const DashboardRoute: React.FC<DashboardRouteProps> = ({
+  path,
+  component,
+  menuItems,
+}) => (
+  <DashboardLayout
+    menuItems={menuItems}
+    profileData={mockProfileData}
+    siteLocations={siteLocations}
+  >
```

Comment:
This is the actual prop provider. These props are passing through their children and so on. I'm not so experienced and didn't use it but using Context API can solve the prop passing problem.